src/serializers/config.rs

```diff
@@ -18,29 +23,52 @@ use super::errors::py_err_se_err;
 #[allow(clippy::struct_field_names)]
 pub(crate) struct SerializationConfig {
     pub timedelta_mode: TimedeltaMode,
+    pub temporal_mode: TemporalMode,
+    prefer_timedelta_mode: bool,
```

Comment:
Im not super happy with this code being repeated here & in timedelta.rs... but im not sure if theres a better way of doing things.

Reply:
Here I would suggest having `timedelta_mode` be `Option<TimedeltaMode>`, which should make it obvious what's going on.